tests/cephadm/test_label_no_schedule.py

```diff
@@ -0,0 +1,55 @@
+import time
+from json import loads
+
+from ceph.ceph_admin import CephAdmin
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class LabelNoScheduleError(Exception):
+    pass
+
+
+def check_daemons(instance, node, daemon):
+    """
+    Method checks all the deamons in node
+    Args:
+        instance: cluster instance
+        node: cluster node
+        daemon: demon name
+    """
+    log.info(f"Checking all the deamons in {node}")
+    cmd = f"cephadm shell ceph orch ps {node} --format json"
+    out, err = instance.installer.exec_command(cmd, sudo=True)
+    all_running_daemons = [data["daemon_type"] for data in loads(out)]
+    if all(item == "osd" for item in all_running_daemons):
+        log.info("All daemons except OSD moved/removed in node")
+        return 0
+    return 1
+
+
+def run(ceph_cluster, **kw):
+    """verify "__no_schedule" label
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+
+    """
+    installer_node = ceph_cluster.get_nodes(role="installer")
+    config = kw.get("config")
+    instance = CephAdmin(cluster=ceph_cluster, **config)
+    node = instance.cluster.get_nodes()[1]
+    config["node"] = node.hostname
+    config["label"] = "_no_schedule"
+    daemon = "osd"
+    # Adding "__no_schedule" label and checking daemons to node.
+    result = CephAdm(installer_node).ceph.orch.label.add(**config)
+    if "Added label" not in result[0]:
+        raise LabelNoScheduleError("Failed to add _no_schedule label to node")
+    time.sleep(60)
```

Comment:
Why constant sleep? Check for some event

Reply:
Label "no_schedule" : Remove all the daemons in particular host except osd.

After running orch command to add label "no_schedule" cephadm takes some time to remove all the daemons except osd.
Time varies cluster to cluster but not exceed more than 60 sec.

Reply:
@MohitBis Use waiter function as suggested.

Reply:
Updated